packages/mobile/src/identity/verification.ts

```diff
@@ -153,28 +153,39 @@ export function* doVerificationFlow() {
     // Mark codes completed in previous attempts
     yield put(completeAttestationCode(NUM_ATTESTATIONS_REQUIRED - status.numAttestationsRemaining))
 
-    // Request any additional attestations needed to be verified
-    yield call(
-      requestNeededAttestations,
+    // The set of attestations we can reveal right now
+    let attestations: ActionableAttestation[] = yield call(
+      getActionableAttestations,
       attestationsContract,
-      stableTokenContract,
-      status.numAttestationsRequestsNeeded,
       e164NumberHash,
       account
     )
 
+    while (attestations.length < status.numAttestationsRemaining) {
+      // Request any additional attestations beyond the original set
+      yield call(
```

Comment:
Is this strategy meant to be a short-term fix or the intended solution? Seems suboptimal to need an indeterminate number of request rounds like this.
If it's short term, please leave a TODO

Reply:
There is optimization potential in here to avoid refetching existing attestations we already know should be good, but fundamentally this is the intended solution